copy/copy.go

```diff
@@ -1234,6 +1242,10 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 	isCompressed := decompressor != nil
 	destStream = bar.ProxyReader(destStream)
 
+	if expectedCompressionFormat, known := expectedCompressionFormats[srcInfo.MediaType]; known && compressionFormat.Name() != expectedCompressionFormat.Name() {
+		logrus.Debugf("blob %s with type %s should be compressed with %s, but compressor appears to be %s", srcInfo.Digest.String(), srcInfo.MediaType, compressionFormat.Name(), expectedCompressionFormat.Name())
```

Comment:
Should this be Warnf?

Reply:
Maybe?

Reply:
Either would make sense.

_Right now_, I’m thinking “What’s the user going to do about it?”
- If the resulting build/copy/use pipeline _works_, the user is not very motivated to do something about that warning, it’s just going to be ignored.
- If the build/copy/use pipeline doesn’t work, the warning could be useful for diagnosis, but (assuming the wrong MIME types are not created by the GitHub.com/containers codebase, which is a bit of a hubris, sure), really the debugging process will start with the error reported by the consumer, not by an optional copy tool in the middle.
- We AFAIK provide no tools/process to recover from such a mismatch, apart maybe from some pull/push sequences that might repair the MIME types purely by side effect, not as an intentional maintained/tested feature.

So in my view this is closer to the “most users won’t care or can’t act on it” side, i.e. debug logs, than to the “most users need to do something definitve about this with some urgency”, i.e. a public warning. But that’s a very weak judgement and opinions can certainly differ.